### ISSUE TITLE
How to generate video?

### ISSUE THREAD

**Issue by dialogueeeeee** [ashawkey/torch-ngp#87]
Your work is very impressive. The results of running the code are all pictures. I want to know how to **generate video**? Is it OK to simply splice the pictures together?

**Comment by ashawkey** [comment]
@dialogueeeeee Yes, an example by replacing the `test` function in `utils.py`:
```python
def test(self, loader, save_path=None, name=None, write_image=False):

        if save_path is None:
            save_path = os.path.join(self.workspace, 'results')

        if name is None:
            name = f'{self.name}_ep{self.epoch:04d}'

        os.makedirs(save_path, exist_ok=True)
        
        self.log(f"==> Start Test, save results to {save_path}")

        pbar = tqdm.tqdm(total=len(loader) * loader.batch_size, bar_format='{percentage:3.0f}% {n_fmt}/{total_fmt} [{elapsed}<{remaining}, {rate_fmt}]')
        self.model.eval()

        all_preds = []

        with torch.no_grad():

            for i, data in enumerate(loader):
                
                with torch.cuda.amp.autocast(enabled=self.fp16):
                    preds, preds_depth = self.test_step(data)                
                
                path = os.path.join(save_path, f'{name}_{i:04d}.png')
                path_depth = os.path.join(save_path, f'{name}_{i:04d}_depth.png')

                #self.log(f"[INFO] saving test image to {path}")

                if self.opt.color_space == 'linear':
                    preds = linear_to_srgb(preds)

                pred = preds[0].detach().cpu().numpy()
                pred = (pred * 255).astype(np.uint8)

                pred_depth = preds_depth[0].detach().cpu().numpy()
                pred_depth = (pred_depth * 255).astype(np.uint8)

                if write_image:
                    imageio.imwrite(path, pred)
                    imageio.imwrite(path_depth, pred_depth)

                all_preds.append(pred)

                pbar.update(loader.batch_size)

        # write video
        all_preds = np.stack(all_preds, axis=0)
        imageio.mimwrite(os.path.join(save_path, f'{name}.mp4'), all_preds, fps=25, quality=10, macro_block_size=1)

        self.log(f"==> Finished Test.")
```

**Comment by dialogueeeeee** [comment]
Thank you so much for your code! Is it the same in **D-Nerf** to generate a video?

**Comment by ashawkey** [comment]
Yes, just use the `mimwrite` from `imageio`.

**Comment by dialogueeeeee** [comment]
Hi, Thank you so much for your code! I have changed the `test` function in `nerf/utils.py`, but I still have the following problems when testing both on **Nerf** and **D-Nerf**:
- For Nerf, when I exported the `.mp4` file obtained from the test, I found that it was a 0-second video without any content.
- For D-Nerf, it will not generate `.mp4` files. Maybe it doesn't call the `test` function in `nerf/utils.py`
